run.sh

```diff
@@ -2,6 +2,39 @@
 
 set -ex
 
+# Ensure we're running on a Raspberry Pi
+if ! grep -q "Raspberry Pi" /proc/device-tree/model 2>/dev/null; then
```

Comment:
@abe-winter if you could take a lot at just this file, there was an instance where cargo wasn't sy,linked properly, it seems to be installed on every raspberry pi and this is a raspberry pi module only, so I'd like to deal with that here in the run.sh, and/or a first_run script in the future if I ever move back to pyinstaller.

Reply:
- guessing cargo link is only required for uv, won't be needed in pyinstaller case
- note that you can `pip install uv` instead of `curl astral | sh` if you know python is available on the device